release_notes/ocp-4-6-release-notes.adoc

```diff
@@ -2167,6 +2167,8 @@ After upgrading, it is recommended to disable the `NET_RAW` and `SYS_CHROOT` cap
 *Important*: If you are upgrading from an earlier release, upgrade to 4.5.16 before you upgrade to 4.6.
 (link:https://bugzilla.redhat.com/show_bug.cgi?id=1874671[*BZ#1874671*]).
 
+* The {product-title} machine-api bare metal actuator currently deletes machine objects when the underlying host is deleted. This behavior does not align with other cloud provider actuators and could result in the replacement machine not coming available. As a workaround, find the BareMetalHost that corresponds with the machine and remove it. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1868104[*BZ#1868104*])
```

Comment:
Sorry to be the bearer of bad news, but this isn't correct. The "underlying host" in the first sentence refers to the same BareMetalHost object as in the workaround, so it's completely circular.

The important part of the workaround I linked was to annotate the Machine as the one to be deleted (note: it's a different annotation in OpenShift from in the upstream doc I linked) before deleting the host and then proactively scale the MachineSet down afterwards. This also has the effect of deleting the Machine, the only difference is that you are expecting it. TBH I'm not even sure this is a good idea, because it's timing-dependent (if the Machine deletes itself and gets replaced before you scale down, the Machine that gets removed on scale down will be selected at random by default, so it may not be the newly-created not-yet-provisioned one).

The truth is that this just isn't a very interesting thing to call out for users. It works differently from other platforms in that if the underlying provider (in this case a BareMetalHost object in k8s - for e.g. AWS it would just be an EC2 instance) is deleted by the user, the Machine object disappears, whereas on other platforms it doesn't. That might be surprising for people extremely familiar with other plaforms, and could make debugging trickier. There's no workaround per se, in the sense that there's no way to stop the Machine from going away.

Deleting the Machine does mean that it will be replaced by the MachineSet controller in case you scale down, and probably a small book could be written about the ways that this will subtly change the outcomes depending on how other things in the system are configured, but it's not clear that anyone would do anything differently after reading it. For example, if you have other unused BareMetalHosts available then a replacement will get provisioned (possibly unexpectedly!), but if you don't then there'll just be an extra Machine in the `provisioning` phase (instead of an extra Machine in the `failed` phase as on other platforms). But if you have the thing enabled that scales the MachineSet to match the number of BareMetalHosts available, then the Machine would have been deleted anyway. Or if you have MachineHealthChecks enabled then the Machine will get deleted on any platform.

If there's anything useful to be said at all about this, it's just the first part IMHO:

>  * The {product-title} machine-api baremetal actuator currently deletes Machine objects when the underlying BareMetalHost is deleted. This behavior does not align with other cloud provider actuators, which move a Machine object to the `failed` phase rather than removing it altogether if the underlying cloud provider resource is deleted.
> (link:https://bugzilla.redhat.com/show_bug.cgi?id=1868104[*BZ#1868104*])

Reply:
@zaneb thank you for the detailed explanation. I have updated the known issue description to closely mirror your recommendation. It's a bit clearer now for me, and I agree that makes sense. I will move forward with this so we can get this into the release notes. Thanks again!